File: src/main/java/javabot/commands/ListChannels.java
package javabot.commands;

import java.util.List;
import java.util.ArrayList;

import javabot.BotEvent;
import javabot.Javabot;
import javabot.Message;
import javabot.model.Channel;
import javabot.dao.ChannelDao;
import javabot.dao.util.QueryParam;
import org.springframework.beans.factory.annotation.Autowired;

/**
 * Created Dec 17, 2008
 *
 * @author <a href="mailto:jlee@antwerkz.com">Justin Lee</a>
 */
public class ListChannels implements Command {
    @Autowired
    private ChannelDao dao;

    @Override
    public List<Message> execute(Javabot bot, BotEvent event, List<String> args) {
        List<Message> messages = new ArrayList<Message>();
        List<Channel> channels = dao.find(new QueryParam(0, Integer.MAX_VALUE));
        messages.add(new Message(event.getChannel(), event, event.getSender() + ", I'll list the channels in a"
            + " private message for you"));
        for (Channel channel : channels) {
            messages.add(new Message(event.getSender(), event, channel.getName() +
                (channel.getLogged() ? "(logged)" : "")));
        }
        return messages;
    }
}

File: src/main/kotlin/javabot/commands/ListChannels.kt
package javabot.commands

import com.antwerkz.sofia.Sofia
import jakarta.inject.Inject
import java.lang.String.format
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao
import javabot.dao.ChannelDao
import javabot.dao.util.QueryParam

class ListChannels
@Inject
constructor(bot: Javabot, adminDao: AdminDao, var channelDao: ChannelDao) :
    AdminCommand(bot, adminDao) {

    override fun execute(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        val channels = channelDao.find(QueryParam(0, Integer.MAX_VALUE))
        responses.add(Message(event, Sofia.adminListChannelsPreamble(event.user.nick)))
        val names =
            channels.joinToString(
                ", ",
                transform = { channel ->
                    format("%s %s", channel.name, if (channel.logged) "(logged)" else "")
                }
            )
        responses.add(Message(event.user, names))

        return responses
    }
}
